### PR TITLE
dbclean is now using the new database functions

### PR DESCRIPTION
**Pull request by annando**
Additionally there are new functions for commit, rollback and transaction.

### CODE REVIEW

File: include/dba.php [hunk 1]
```diff
@@ -872,10 +916,11 @@ static public function delete($table, $param, $in_commit = false, &$callstack =
 			}
 		}
 
-		if (!$in_commit) {
+		if (!$in_process) {
 			// Now we finalize the process
-			self::p("COMMIT");
-			self::p("START TRANSACTION");
+			if (!self::$in_transaction) {
```
Comment (Hypolite) [inline include/dba.php]: I think this test for `$in_transaction` should be inside `transaction()`.

File: include/dba.php [hunk 2]
```diff
@@ -915,13 +962,17 @@ static public function delete($table, $param, $in_commit = false, &$callstack =
 						logger(dba::replace_parameters($sql, $field_values), LOGGER_DATA);
 
 						if (!self::e($sql, $field_values)) {
-							self::p("ROLLBACK");
+							if (!self::$in_transaction) {
```
Comment (Hypolite) [inline include/dba.php]: I think this test for `$in_transaction` should be inside `rollback()`. Additionally, I would expect a positive check for `$in_transaction`, is the negative check intended?

File: include/dba.php [hunk 3]
```diff
@@ -915,13 +962,17 @@ static public function delete($table, $param, $in_commit = false, &$callstack =
...
+								self::rollback();
+							}
 							return false;
 						}
 					}
 				}
 			}
-			self::p("COMMIT");
+			if (!self::$in_transaction) {
```
Comment (Hypolite) [inline include/dba.php]: I think this test for `$in_transaction` should be inside `commit()`. Additionally, I would expect a positive check for `$in_transaction`, is the negative check intended?

File: include/poller.php [hunk 4]
```diff
@@ -128,26 +128,26 @@ function poller_execute($queue) {
 
 	if (!$upd) {
 		logger("Couldn't update queue entry ".$queue["id"]." - skip this execution", LOGGER_DEBUG);
-		q("COMMIT");
+		dba::commit();
```
Comment (Hypolite) [inline include/poller.php]: What is the purpose of those commits without a transaction start?

File: include/dba.php (hunk 3 above)
Comment (annando) [inline include/dba.php]: The ```delete``` command could be part of bigger transaction.

File: include/dba.php (hunk 1 above)
Comment (annando) [inline include/dba.php]: The ```delete``` command could be part of bigger transaction.

File: include/dba.php (hunk 2 above)
Comment (annando) [inline include/dba.php]: The ```delete``` command could be part of bigger transaction.

File: include/poller.php (hunk 4 above)
Comment (annando) [inline include/poller.php]: There is a transaction start in the poller.php as well.

File: include/dba.php (hunk 3 above)
Comment (Hypolite) [inline include/dba.php]: Ha, I see now. Thanks for the clarification.

**Pull request merged**